doxygen/man/man3/cMainWindow.3
.TH "cMainWindow" 3 "Thu Feb 11 2016" "Kooky" \" -*- nroff -*-
.ad l
.nh
.SH NAME
cMainWindow \- This is the class that implements the main window\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <cmainwindow\&.h>\fP
.PP
Inherits QMainWindow\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBcMainWindow\fP (QWidget *parent=0)"
.br
.RI "\fIConstructor\&. \fP"
.ti -1c
.RI "\fB~cMainWindow\fP ()"
.br
.RI "\fIDestructor\&. \fP"
.ti -1c
.RI "QList< \fBcPlugin\fP * > \fBpluginList\fP ()"
.br
.RI "\fIReturns the list of loaded plugins\&. \fP"
.in -1c
.SS "Private Slots"

.in +1c
.ti -1c
.RI "void \fBpluginImportTriggered\fP ()"
.br
.RI "\fITesting of import trigger raised by import plugins\&. \fP"
.ti -1c
.RI "void \fBpluginExportTriggered\fP ()"
.br
.RI "\fITesting of export trigger raised by export plugins\&. \fP"
.ti -1c
.RI "void \fBpluginDBTriggered\fP ()"
.br
.RI "\fITesting of database trigger raised by database connector plugins\&. \fP"
.ti -1c
.RI "void \fBon_m_lpMenu_File_Exit_triggered\fP ()"
.br
.RI "\fIMenu entry 'File -> Exit' triggered\&. \fP"
.ti -1c
.RI "void \fBon_m_lpMenu_Tools_Options_triggered\fP ()"
.br
.RI "\fIMenu entry 'Tools -> Options' triggered\&. \fP"
.ti -1c
.RI "void \fBonIngredientsListCustomContextMenu\fP (const QPoint &point)"
.br
.RI "\fICustom contest menu of ingredients list has been requested\&. \fP"
.ti -1c
.RI "void \fBonIngredientsListNew\fP ()"
.br
.RI "\fINew ingredient has been requested\&. \fP"
.ti -1c
.RI "void \fBonIngredientsListImport\fP ()"
.br
.RI "\fIImport of ingredient has been requested\&. \fP"
.ti -1c
.RI "void \fBonIngredientsListDelete\fP ()"
.br
.RI "\fIDelete of ingredient has been requested\&. \fP"
.ti -1c
.RI "void \fBonIngredientsListEdit\fP ()"
.br
.RI "\fIEdit of ingredient has been requested\&. \fP"
.in -1c
.SS "Private Member Functions"

.in +1c
.ti -1c
.RI "void \fBinit\fP ()"
.br
.RI "\fIInitialized main windows\&. \fP"
.ti -1c
.RI "void \fBinitDefaultSettings\fP ()"
.br
.RI "\fIInitializes default settings for Kooky\&. \fP"
.ti -1c
.RI "void \fBloadPlugins\fP (const QString &szPluginDir)"
.br
.RI "\fILoads plugins in the given path\&. \fP"
.ti -1c
.RI "\fBcPlugin\fP * \fBplugin\fP (QAction *lpAction)"
.br
.RI "\fIFinds a plugin with the given action\&. \fP"
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "Ui::cMainWindow * \fBui\fP"
.br
.ti -1c
.RI "QList< \fBcPlugin\fP * > \fBm_pluginList\fP"
.br
.ti -1c
.RI "QStandardItemModel * \fBm_lpIngredientsListModel\fP"
.br
.ti -1c
.RI "QAction * \fBm_lpIngredientsListNew\fP"
.br
.ti -1c
.RI "QAction * \fBm_lpIngredientsListImport\fP"
.br
.ti -1c
.RI "QAction * \fBm_lpIngredientsListDelete\fP"
.br
.ti -1c
.RI "QAction * \fBm_lpIngredientsListEdit\fP"
.br
.ti -1c
.RI "QMenu * \fBm_lpIngredientsListMenu\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
This class implements the main window and main functionality of Kooky\&.
.PP
\fBAuthor:\fP
.RS 4
Herwig Birke
.RE
.PP
\fBVersion:\fP
.RS 4
1\&.0
.RE
.PP
\fBDate:\fP
.RS 4
2016/02/09 
.RE
.PP

.SH "Constructor & Destructor Documentation"
.PP 
.SS "cMainWindow::cMainWindow (QWidget * parent = \fC0\fP)\fC [explicit]\fP"

.PP
\fBParameters:\fP
.RS 4
\fIparent\fP 
.RE
.PP

.PP
References init()\&.
.SS "cMainWindow::~cMainWindow ()"

.PP
References m_lpIngredientsListDelete, m_lpIngredientsListEdit, m_lpIngredientsListNew, and ui\&.
.SH "Member Function Documentation"
.PP 
.SS "void cMainWindow::init ()\fC [private]\fP"

.PP
References initDefaultSettings(), loadPlugins(), m_lpIngredientsListDelete, m_lpIngredientsListEdit, m_lpIngredientsListImport, m_lpIngredientsListMenu, m_lpIngredientsListModel, m_lpIngredientsListNew, onIngredientsListCustomContextMenu(), onIngredientsListDelete(), onIngredientsListEdit(), onIngredientsListImport(), onIngredientsListNew(), and ui\&.
.PP
Referenced by cMainWindow()\&.
.SS "void cMainWindow::initDefaultSettings ()\fC [private]\fP"

.PP
Referenced by init()\&.
.SS "void cMainWindow::loadPlugins (const QString & szPluginDir)\fC [private]\fP"

.PP
\fBParameters:\fP
.RS 4
\fIszPluginDir\fP Path to plugins 
.RE
.PP

.PP
References cPlugin::isValid(), m_pluginList, pluginDBTriggered(), pluginExportTriggered(), pluginImportTriggered(), cPlugin::pluginName(), cPlugin::setAction(), and ui\&.
.PP
Referenced by init()\&.
.SS "void cMainWindow::on_m_lpMenu_File_Exit_triggered ()\fC [private]\fP, \fC [slot]\fP"

.SS "void cMainWindow::on_m_lpMenu_Tools_Options_triggered ()\fC [private]\fP, \fC [slot]\fP"

.SS "void cMainWindow::onIngredientsListCustomContextMenu (const QPoint & point)\fC [private]\fP, \fC [slot]\fP"

.PP
\fBParameters:\fP
.RS 4
\fIpoint\fP Current position of mouse cursor\&. 
.RE
.PP

.PP
References m_lpIngredientsListDelete, m_lpIngredientsListEdit, m_lpIngredientsListMenu, m_lpIngredientsListNew, and ui\&.
.PP
Referenced by init()\&.
.SS "void cMainWindow::onIngredientsListDelete ()\fC [private]\fP, \fC [slot]\fP"

.PP
Referenced by init()\&.
.SS "void cMainWindow::onIngredientsListEdit ()\fC [private]\fP, \fC [slot]\fP"

.PP
Referenced by init()\&.
.SS "void cMainWindow::onIngredientsListImport ()\fC [private]\fP, \fC [slot]\fP"

.PP
References m_pluginList, and cImportIngredientDialog::setPluginList()\&.
.PP
Referenced by init()\&.
.SS "void cMainWindow::onIngredientsListNew ()\fC [private]\fP, \fC [slot]\fP"

.PP
Referenced by init()\&.
.SS "\fBcPlugin\fP * cMainWindow::plugin (QAction * lpAction)\fC [private]\fP"

.PP
\fBParameters:\fP
.RS 4
\fIlpAction\fP Action of the plugin\&. 
.RE
.PP
\fBReturns:\fP
.RS 4
\fBcPlugin\fP Plugin corresponding to the action, NULL if not found\&. 
.RE
.PP

.PP
References m_pluginList\&.
.PP
Referenced by pluginDBTriggered()\&.
.SS "void cMainWindow::pluginDBTriggered ()\fC [private]\fP, \fC [slot]\fP"

.PP
References cInterface::config(), cPlugin::dbInterface(), cDBInterface::init(), plugin(), cInterface::pluginName(), and cPlugin::pluginName()\&.
.PP
Referenced by loadPlugins()\&.
.SS "void cMainWindow::pluginExportTriggered ()\fC [private]\fP, \fC [slot]\fP"

.PP
Referenced by loadPlugins()\&.
.SS "void cMainWindow::pluginImportTriggered ()\fC [private]\fP, \fC [slot]\fP"

.PP
Referenced by loadPlugins()\&.
.SS "QList< \fBcPlugin\fP * > cMainWindow::pluginList ()"

.PP
\fBReturns:\fP
.RS 4
QList<cPlugin *> List of loaded plugins\&. 
.RE
.PP

.PP
References m_pluginList\&.
.PP
Referenced by cOptions::cOptions()\&.
.SH "Member Data Documentation"
.PP 
.SS "QAction* cMainWindow::m_lpIngredientsListDelete\fC [private]\fP"
'Delee Ingredient' Action for ingredients context menu 
.PP
Referenced by init(), onIngredientsListCustomContextMenu(), and ~cMainWindow()\&.
.SS "QAction* cMainWindow::m_lpIngredientsListEdit\fC [private]\fP"
'Edit Ingredient' Action for ingredients context menu 
.PP
Referenced by init(), onIngredientsListCustomContextMenu(), and ~cMainWindow()\&.
.SS "QAction* cMainWindow::m_lpIngredientsListImport\fC [private]\fP"
'Import Ingredient' Action for ingredients context menu 
.PP
Referenced by init()\&.
.SS "QMenu* cMainWindow::m_lpIngredientsListMenu\fC [private]\fP"
Context menu for ingredients list 
.PP
Referenced by init(), and onIngredientsListCustomContextMenu()\&.
.SS "QStandardItemModel* cMainWindow::m_lpIngredientsListModel\fC [private]\fP"
standard item model for ingredients list 
.PP
Referenced by init()\&.
.SS "QAction* cMainWindow::m_lpIngredientsListNew\fC [private]\fP"
'New Ingredient' Action for ingredients context menu 
.PP
Referenced by init(), onIngredientsListCustomContextMenu(), and ~cMainWindow()\&.
.SS "QList<\fBcPlugin\fP*> cMainWindow::m_pluginList\fC [private]\fP"
List of loaded plugins 
.PP
Referenced by loadPlugins(), onIngredientsListImport(), plugin(), and pluginList()\&.
.SS "Ui::cMainWindow* cMainWindow::ui\fC [private]\fP"
the main window GUI 
.PP
Referenced by init(), loadPlugins(), onIngredientsListCustomContextMenu(), and ~cMainWindow()\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for Kooky from the source code\&.


Kooky/cmainwindow.cpp
#include "cmainwindow.h"
#include "ui_cmainwindow.h"
#include "cimportinterface.h"
#include "cingredient.h"

#include "typedef.h"

#include "coptions.h"

#include "cimportingredientdialog.h"

#include "cmessageanimatedialog.h"

#include <QPluginLoader>
#include <QDebug>
#include <QMessageBox>
#include <QSettings>
#include <QDir>
#include <QImage>
#include <QContextMenuEvent>
#include <QThread>


cMainWindow::cMainWindow(QWidget *parent) :
	QMainWindow(parent),
	ui(new Ui::cMainWindow),
	m_lpIngredientsListModel(0),
	m_lpIngredientsListNew(0),
	m_lpIngredientsListImport(0),
	m_lpIngredientsListDelete(0),
	m_lpIngredientsListEdit(0),
	m_lpIngredientsListMenu(0),
	m_lpDB(0)
{
	QThread::msleep(2000);
	init();
}

cMainWindow::~cMainWindow()
{
	if(m_lpIngredientsListNew)
		delete m_lpIngredientsListNew;
	if(m_lpIngredientsListDelete)
		delete m_lpIngredientsListDelete;
	if(m_lpIngredientsListEdit)
		delete m_lpIngredientsListEdit;
	delete ui;
}

void cMainWindow::init()
{
	QSettings	settings;

	if(settings.value("program/version", QVariant(-1.0)).toDouble() == -1.0)
		initDefaultSettings();

	ui->setupUi(this);

	loadPlugins("plugins");

	m_lpIngredientsListModel	= new QStandardItemModel(0, 1);
	ui->m_lpIngredientsList->setModel(m_lpIngredientsListModel);

	connect(ui->m_lpMenu_File_Exit, SIGNAL(triggered(bool)), this, SLOT(menuFileExitTriggered()));
	connect(ui->m_lpMenu_Tools_Options, SIGNAL(triggered(bool)), this, SLOT(menuToolsOptionsTriggered()));

	ui->m_lpIngredientsList->setContextMenuPolicy(Qt::CustomContextMenu);
	connect(ui->m_lpIngredientsList, SIGNAL(customContextMenuRequested(const QPoint &)), this, SLOT(ingredientsListCustomContextMenu(const QPoint &)));
	m_lpIngredientsListMenu		= new QMenu(ui->m_lpIngredientsList);

	m_lpIngredientsListNew		= new QAction("&New", m_lpIngredientsListMenu);
	m_lpIngredientsListMenu->addAction(m_lpIngredientsListNew);
	connect(m_lpIngredientsListNew, SIGNAL(triggered(bool)), this, SLOT(ingredientsListNewTriggered()));

	m_lpIngredientsListImport	= new QAction("&Import", m_lpIngredientsListMenu);
	m_lpIngredientsListMenu->addAction(m_lpIngredientsListImport);
	connect(m_lpIngredientsListImport, SIGNAL(triggered(bool)), this, SLOT(ingredientsListImportTriggered()));

	m_lpIngredientsListDelete	= new QAction("&Delete", m_lpIngredientsListMenu);
	m_lpIngredientsListMenu->addAction(m_lpIngredientsListDelete);
	connect(m_lpIngredientsListDelete, SIGNAL(triggered(bool)), this, SLOT(ingredientsListDeleteTriggered()));

	m_lpIngredientsListEdit		= new QAction("&Edit", m_lpIngredientsListMenu);
	m_lpIngredientsListMenu->addAction(m_lpIngredientsListEdit);
	connect(m_lpIngredientsListEdit, SIGNAL(triggered(bool)), this, SLOT(ingredientsListEditTriggered()));

	bool	bConnected	= false;

	while(!bConnected)
	{
		if(!m_lpDB)
		{
			if(QMessageBox::critical(this, "Warning", tr("Database can't be opened. Do you want to configure plugins?"), QMessageBox::Yes | QMessageBox::No) == QMessageBox::No)
			{
				QTimer::singleShot(0, this, SLOT(menuFileExitTriggered()));
				return;
			}

			menuToolsOptionsTriggered();
		}
		else if(!m_lpDB->dbInterface())
		{
			if(QMessageBox::critical(this, "Warning", tr("Database can't be opened. Do you want to configure plugins?"), QMessageBox::Yes | QMessageBox::No) == QMessageBox::No)
			{
				QTimer::singleShot(0, this, SLOT(menuFileExitTriggered()));
				return;
			}

			menuToolsOptionsTriggered();
		}
		else if(!m_lpDB->dbInterface()->connect())
		{
			if(QMessageBox::critical(this, "Warning", tr("Database can't be opened. Do you want to configure plugins?"), QMessageBox::Yes | QMessageBox::No) == QMessageBox::No)
			{
				QTimer::singleShot(0, this, SLOT(menuFileExitTriggered()));
				return;
			}

			menuToolsOptionsTriggered();
		}
		else
			bConnected	= true;
	}
	loadIngredients();
}

void cMainWindow::initDefaultSettings()
{
	QSettings	settings;
	QDir		dir;

	settings.setValue("program/version", QVariant(0.1));
	settings.setValue("program/dataDir", QVariant(QString("%1/.kooky").arg(QDir::homePath())));
	settings.setValue("program/pluginDir", QVariant(QString("%1/.kooky/plugins").arg(QDir::homePath())));
	settings.setValue("program/receiptDir", QVariant(QString("%1/.kooky/receipts").arg(QDir::homePath())));
	settings.setValue("program/ingredientDir", QVariant(QString("%1/.kooky/ingredients").arg(QDir::homePath())));
	settings.setValue("program/imageDir", QVariant(QString("%1/.kooky/images").arg(QDir::homePath())));

	dir.mkpath(settings.value("program/dataDir").toString());
	dir.mkpath(settings.value("program/pluginDir").toString());
	dir.mkpath(settings.value("program/receiptDir").toString());
	dir.mkpath(settings.value("program/ingredientDir").toString());
	dir.mkpath(settings.value("program/imageDir").toString());
}

void cMainWindow::loadPlugins(const QString& szPluginDir)
{
	QCoreApplication::addLibraryPath(szPluginDir);
	QDir		pluginDir(szPluginDir);
	QSettings	settings;

#ifdef Q_OS_WIN
	QStringList	szImportList	= pluginDir.entryList(QStringList() << "i*.dll");
	QStringList	szExportList	= pluginDir.entryList(QStringList() << "e*.dll");
	QStringList	szDBList		= pluginDir.entryList(QStringList() << "d*.dll");
#else
	QStringList	szImportList	= pluginDir.entryList(QStringList() << "libi*.so");
	QStringList	szExportList	= pluginDir.entryList(QStringList() << "libe*.so");
	QStringList	szDBList		= pluginDir.entryList(QStringList() << "libd*.so");
#endif

	for(int z = 0;z < szImportList.count();z++)
	{
#ifdef Q_OS_WIN
		QString		szFile	= szImportList.at(z);
#else
		QString		szFile	= szImportList.at(z).mid(3, szImportList.at(z).length()-6);
#endif
		cPlugin*	lpPlugin	= new cPlugin(szFile);
		if(lpPlugin->isValid())
		{
			QAction*	lpAction	= new QAction(this);
			ui->m_lpMenu_Plugins_Import->addAction(lpAction);
			lpAction->setText(lpPlugin->pluginName());
			connect(lpAction, SIGNAL(triggered()), this, SLOT(pluginImportTriggered()));
			lpPlugin->setAction(lpAction);
			m_pluginList.append(lpPlugin);
		}
	}

	for(int z = 0;z < szExportList.count();z++)
	{
#ifdef Q_OS_WIN
		QString		szFile	= szExportList.at(z);
#else
		QString		szFile	= szExportList.at(z).mid(3, szExportList.at(z).length()-6);
#endif
		cPlugin*	lpPlugin	= new cPlugin(szFile);
		if(lpPlugin->isValid())
		{
			QAction*	lpAction	= new QAction(this);
			ui->m_lpMenu_Plugins_Export->addAction(lpAction);
			lpAction->setText(lpPlugin->pluginName());
			connect(lpAction, SIGNAL(triggered()), this, SLOT(pluginExportTriggered()));
			lpPlugin->setAction(lpAction);
			m_pluginList.append(lpPlugin);
		}
	}

	for(int z = 0;z < szDBList.count();z++)
	{
#ifdef Q_OS_WIN
		QString		szFile	= szDBList.at(z);
#else
		QString		szFile	= szDBList.at(z).mid(3, szDBList.at(z).length()-6);
#endif
		cPlugin*	lpPlugin	= new cPlugin(szFile);
		if(lpPlugin->isValid())
		{
			QAction*	lpAction	= new QAction(this);
			ui->m_lpMenu_Plugins_Database->addAction(lpAction);
			lpAction->setText(lpPlugin->pluginName());
			connect(lpAction, SIGNAL(triggered()), this, SLOT(pluginDBTriggered()));
			lpPlugin->setAction(lpAction);
			m_pluginList.append(lpPlugin);
			if(settings.value(QString("plugins/%1/enabled").arg(lpPlugin->pluginName()), QVariant(false)).toBool())
				m_lpDB	= lpPlugin;
		}
	}
}

cPlugin* cMainWindow::plugin(QAction* lpAction)
{
	for(int z = 0;z < m_pluginList.count();z++)
	{
		if(m_pluginList.at(z)->action() == lpAction)
			return(m_pluginList.at(z));
	}
	return(0);
}

QList<cPlugin*> cMainWindow::pluginList()
{
	return(m_pluginList);
}

void cMainWindow::loadIngredients()
{
	m_lpIngredientsListModel->removeRows(0, m_lpIngredientsListModel->rowCount());

	QString			szOldGroup("");
	QStandardItem*	lpGroupItem	= 0;

	if(m_lpDB)
	{
		cDBInterface* lpInterface	= m_lpDB->dbInterface();
		if(lpInterface)
		{
			INGREDIENT_LIST	ingredients	= lpInterface->ingredients();
			for(int z = 0;z < ingredients.count();z++)
			{
				if(ingredients.at(z).szGroup != szOldGroup)
				{
					szOldGroup	= ingredients.at(z).szGroup;
					lpGroupItem	= new QStandardItem(szOldGroup);
					m_lpIngredientsListModel->appendRow(lpGroupItem);
				}
				QStandardItem*	lpNew	= new QStandardItem(ingredients.at(z).szIngredient);
				lpGroupItem->appendRow(lpNew);
			}
		}
	}
	ui->m_lpIngredientsList->expandAll();
}

void cMainWindow::pluginImportTriggered()
{
/*
	QSettings	settings;
	cIngredient	ingredient(QString("%1/000001.ingredient").arg(settings.value("program/ingredientDir").toString()));
	ingredient.save();
*/
/*
	QAction*	lpAction	= qobject_cast<QAction *>(sender());
	if(lpAction)
	{
		cPlugin*	lpPlugin	= plugin(lpAction);
		if(lpPlugin)
		{
			cImportInterface*	lpImportInterface	= lpPlugin->importInterface();
			QString				str					= lpImportInterface->pluginName();
			if(lpImportInterface)
			{
				QStringList	list	= lpImportInterface->search("kuhmilch", "bla");

				if(lpImportInterface->load(0))
				{
					qreal		v;
					QString		str;

					str	= lpImportInterface->group(cIngredient::iIngredientCarbohydrates);
					str	= lpImportInterface->name(cIngredient::iIngredientCarbohydrates);
					v	= lpImportInterface->value(cIngredient::iIngredientCarbohydrates);
					v	= lpImportInterface->value(cIngredient::iIngredientBreadUnits);
					v	= lpImportInterface->value(cIngredient::iIngredientWater);
					v	= lpImportInterface->value(cIngredient::iIngredientCholesterol);
					v	= lpImportInterface->value(cIngredient::iIngredientCopper);
					v = 0;
				}
			}
		}
	}
*/
}

void cMainWindow::pluginExportTriggered()
{
}

void cMainWindow::pluginDBTriggered()
{
	QAction*	lpAction	= qobject_cast<QAction *>(sender());
	if(lpAction)
	{
		cPlugin*	lpPlugin	= plugin(lpAction);
		if(lpPlugin)
		{
			if(!lpPlugin->pluginName().compare("MySQL DB"))
			{
				cDBInterface*		lpDBInterface	= lpPlugin->dbInterface();
				QString				str				= lpDBInterface->pluginName();
				if(lpDBInterface)
					lpDBInterface->init();
			}
			else if(!lpPlugin->pluginName().compare("SQLite DB"))
			{
				cDBInterface*		lpDBInterface	= lpPlugin->dbInterface();
				QString				str				= lpDBInterface->pluginName();
				if(lpDBInterface)
					lpDBInterface->init();
			}
		}
	}
}

void cMainWindow::menuFileExitTriggered()
{
	qApp->exit();
}

void cMainWindow::menuToolsOptionsTriggered()
{
	cOptions	options(this);
	if(options.exec() == QDialog::Rejected)
		return;

	QSettings	settings;

	for(int z = 0;z < m_pluginList.count();z++)
	{
		cPlugin*	lpPlugin	= m_pluginList.at(z);
		if(lpPlugin->capability() == cPlugin::PluginCapDB)
		{
			if(settings.value(QString("plugins/%1/enabled").arg(lpPlugin->pluginName()), QVariant(false)).toBool())
			{
				if(lpPlugin != m_lpDB)
				{
					m_lpDB	= lpPlugin;
					m_lpDB->dbInterface()->connect();
					loadIngredients();
					break;
				}
			}
		}
	}
}

void cMainWindow::ingredientsListCustomContextMenu(const QPoint& point)
{
	QModelIndex	index	= ui->m_lpIngredientsList->indexAt(point);
	if(index.isValid())
	{
		m_lpIngredientsListNew->setEnabled(true);
		m_lpIngredientsListDelete->setEnabled(true);
		m_lpIngredientsListEdit->setEnabled(true);
	}
	else
	{
		m_lpIngredientsListNew->setEnabled(true);
		m_lpIngredientsListDelete->setEnabled(false);
		m_lpIngredientsListEdit->setEnabled(false);
	}
	m_lpIngredientsListMenu->exec(ui->m_lpIngredientsList->mapToGlobal(point));
}

void cMainWindow::ingredientsListNewTriggered()
{
}

void cMainWindow::ingredientsListImportTriggered()
{
	cImportIngredientDialog	importIngredientDialog(this);
	importIngredientDialog.setPluginList(m_pluginList);
	if(importIngredientDialog.exec() == QDialog::Rejected)
		return;

	cMessageAnimateDialog*	lpDialog	= new cMessageAnimateDialog(this);
	lpDialog->setTitle("Import Ingredient");
	lpDialog->setMessage("Saving");
	lpDialog->show();

	cIngredient	ingredient	= importIngredientDialog.toIngredient();
	ingredient.save(m_lpDB);
	loadIngredients();

	delete lpDialog;
}

void cMainWindow::ingredientsListDeleteTriggered()
{
}

void cMainWindow::ingredientsListEditTriggered()
{
}
